bin/main.rb

```diff
@@ -1,70 +1,227 @@
 #!/usr/bin/env ruby
-p 'Welcome! This is Tic Tac Toe Game. Have Fun!'
+require './lib/player.rb'
+require './lib/canvas.rb'
 
-p 'Please enter player names'
-player_one = gets.chomp
-puts "#{player_one} you'll play as X"
+# validate and print player choice
+class UserInterface
```

Comment:
- [ ] the main.rb file shouldn't contain classes all classes have to exist within the lib folder, and call them here in a loop 

so let's move the class UserInterface to the lib folder or delete it and move it's methods to the classes you already have.